client/components/ManageBotRunDialog/MarkBotRunFinishedButton/index.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Button } from 'react-bootstrap';
+import { useMutation, useQuery } from '@apollo/client';
+import {
+    SUBMIT_TEST_RESULT_MUTATION,
+    TEST_PLAN_RUN_RESULT_IDS
+} from './queries';
+import { useTriggerLoad } from '../../common/LoadingStatus';
+import { useTestPlanRunValidatedAssertionCounts } from '../../../hooks/useTestPlanRunValidatedAssertionCounts';
+import BasicModal from '../../common/BasicModal';
+import { useTestPlanRunIsFinished } from '../../../hooks/useTestPlanRunIsFinished';
+
+const MarkBotRunFinishedButton = ({ testPlanRun, onClick = () => {} }) => {
+    const {
+        data: testPlanRunCompletionQuery,
+        loading,
+        refetch
+    } = useQuery(TEST_PLAN_RUN_RESULT_IDS, {
+        variables: {
+            testPlanRunId: testPlanRun.id
+        }
+    });
+
+    const { triggerLoad } = useTriggerLoad();
```

Comment:
Here and elsewhere, the `useTriggerLoad` hook doesn't seem useful without a reference to the `loadingMessage` state it provides. It seems like we could forgo it in favor of immediately-invoked function expressions, but I wonder whether we actually want to employ the `LoadingStatus` component for these new asynchronous operations.

Reply:
Good point. I am going to update to use the existing pattern correctly with a LoadingStatus component. I think this pattern would benefit from refactoring app-wide. The loading status could be a singleton in the app so that one doesn't have to remember to include it in children components. The hook could update its state via context.